packages/test/test-end-to-end-tests/src/test/SummarizeFromLatest.spec.ts

```diff
@@ -0,0 +1,224 @@
+        }
+    }
+
+    const loadContainer = async (
+        summaryVersion: string,
+    ): Promise<IContainer> => {
+        const requestHeader = {
+            [LoaderHeader.version]: summaryVersion,
+        };
+        const container = await provider.loadContainer(runtimeFactory, undefined, requestHeader);
+        return container;
+    };
+
+    it("Loading summarizer from old snapshot", async () => {
+        const summarizerClient = await createSummarizer(provider, mainContainer);
+        // 1.
+        let value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0, "1");
+        assert(value === 1, "Value == 1");
+
+        // Generate first Summary
+        const summaryVersion = await waitForSummary(summarizerClient);
+        assert(summaryVersion, "Summary version should be defined");
+
+        // Do some changes
+        value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0);
+        assert(value === 2, "Value == 2");
+
+        // Generate second Summary
+        const summaryVersion1 = await waitForSummary(summarizerClient);
+        assert(summaryVersion1, "Summary version should be defined");
+
+        // Do some changes
+        value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0);
+        assert(value === 3, "Value == 3");
+
+        // Generate third Summary
+        const summaryVersion2 = await waitForSummary(summarizerClient);
+        assert(summaryVersion2, "Summary version should be defined");
+
+        // 4.
+        value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0);
+        assert(value === 4, "Value == 4");
+
+        summarizerClient.close();
+        await provider.ensureSynchronized();
+
+        const newContainer = await loadContainer(summaryVersion1);
+        // Create a summarizer from the first summary.
+        const summarizerClient2 = await createSummarizer(provider, newContainer,
+            summaryVersion1);
+
+        // Do some ops.
+        value = getAndIncrementCellValue(mainDataStore.matrix, 0, 0);
+
+        // Detour getVersions to make sure we fetch once after getting the Summary Ack.
+        let fetchCount: number = 0;
+        const containerRuntime = (summarizerClient2 as any).runtime as ContainerRuntime;
+        let getVersionsFunc = containerRuntime.storage.getVersions;
+        const funcSummary1 = async (versionId: string | null,
+            count: number,
+            scenarioName?: string,
+            fetchSource?: FetchSource,
+        ) => {
+            getVersionsFunc = getVersionsFunc.bind(containerRuntime.storage);
+            const response = await getVersionsFunc(versionId, count, scenarioName, fetchSource);
+            fetchCount++;
+            return response;
+        };
+        containerRuntime.storage.getVersions = funcSummary1;
+
+        // Wait for the summarizer to run again.
+        const summaryVersion3 = await waitForSummary(summarizerClient2);
+        assert(summaryVersion3, "Summary version should be defined");
```

Comment:
Is it possible to validate that the fetch was for the latest snapshot?

Reply:
I could not find an easy way - will proceed with this check in so unblock Jatin and take a look at it later on.